pkg/manager/setconfig.go

```diff
@@ -29,25 +29,20 @@ const SetConfigAlreadyApplied = "Already applied:"
 
 // ValidateNetworkConfig validates the given updates and deletes, according to the path on the configuration
 // for the specified target
-func (m *Manager) ValidateNetworkConfig(configName store.ConfigName, updates change.TypedValueMap,
-	deletes []string) error {
+func (m *Manager) ValidateNetworkConfig(deviceName string, version string,
+	deviceType string, updates change.TypedValueMap, deletes []string) error {
 
-	deviceConfig, _, err := m.getStoredConfig(configName)
+	deviceConfig, _, err := m.getStoredConfig(deviceName, version, deviceType, true)
 	if err != nil {
 		return err
 	}
 	deviceConfigTemporary, _ := store.CreateConfiguration(deviceConfig.Device, deviceConfig.Version,
 		deviceConfig.Type, deviceConfig.Changes)
-	chg, err := m.computeChange(updates, deletes)
-	if err != nil {
-		return err
-	}
-	changeID, err := m.storeChange(chg)
```

Comment:
We were storing the change here - should not have been because it might have been invalid. Now I'm saving it after it's been validated in SetNetworkConfig